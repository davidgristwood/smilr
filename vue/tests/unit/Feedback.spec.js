import { createLocalVue, shallowMount } from '@vue/test-utils'
import Feedback from '@/components/Feedback.vue'
import flushPromises from 'flush-promises'

const localVue = createLocalVue()
import BootstrapVue from 'bootstrap-vue'
localVue.use(BootstrapVue)

jest.mock('@/mixins/api')

describe('Feedback.vue', () => {
  it('shows event data', async () => {
    const wrapper = shallowMount(Feedback, {
      localVue,
      propsData: {
        eventIdProp: 'fake3',
        topicIdProp: '1'
      }
    })

    await flushPromises()
    expect(wrapper).toMatchSnapshot()
  })
})
