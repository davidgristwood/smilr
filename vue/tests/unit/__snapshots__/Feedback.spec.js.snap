// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback.vue shows event data 1`] = `
<div>
  <!---->
  <b-card-stub bodytag="div" titletag="h4" subtitletag="h6" subtitletextvariant="muted" headertag="div" headerbgvariant="primary" headertextvariant="white" footertag="div" tag="div" bordervariant="primary">
    <h2 slot="header" class="text-truncate">
      Lemons (Fake event (current))
    </h2>
    <div class="card-body">
      <h3>Please provide your feedback, click on a face</h3><br>
      <div class="facegroup">
        <face-stub number="1"></face-stub>
        <face-stub number="2"></face-stub>
        <face-stub number="3"></face-stub>
        <face-stub number="4"></face-stub>
        <face-stub number="5"></face-stub>
      </div>
      <!---->
      <div>
        <b-form-textarea-stub value="" placeholder="Any comments (optional)" debounce="0" rows="2" wrap="soft" noresize="true" class="commentbox"></b-form-textarea-stub>
        <!---->
      </div>
      <!---->
    </div>
    <b-modal-stub size="md" centered="true" ignoreenforcefocusselector="" title="Feedback Received" titletag="h5" headerbgvariant="success" hideheaderclose="true" okonly="true" headerclosecontent="&amp;times;" headercloselabel="Close" canceltitle="Cancel" oktitle="All Done" cancelvariant="secondary" okvariant="primary">
      <div class="d-block text-center">
        <h3>Thanks for submitting your feedback! ðŸ˜Š</h3>
      </div>
    </b-modal-stub>
  </b-card-stub>
  <b-modal-stub size="md" centered="true" ignoreenforcefocusselector="" title="Hey! You're keen..." titletag="h5" headerbgvariant="warning" headertextvariant="light" hideheaderclose="true" okonly="true" headerclosecontent="&amp;times;" headercloselabel="Close" canceltitle="Cancel" oktitle="Oh yes, I forgot" cancelvariant="secondary" okvariant="primary">
    <div class="d-block text-center">
      <h3>You've already given feedback for this topic, sorry!</h3>
    </div>
  </b-modal-stub>
</div>
`;
